app/services/movie_service.rb

```diff
@@ -0,0 +1,29 @@
+require 'json'
+
+class MovieService
+
+  def movie_search(query)
+    get_url("/?api_key=#{ENV['movie_api_key']}&query=#{query}&language=en-US")
+  end
+
+  def movie_cast(movie_id)
+    get_url("/movie/#{movie_id}/credits?api_key=#{ENV['movie_api_key']}&language=en-US")
+  end
+
+  def movie_review(movie_id)
+    get_url("/movie/#{movie_id}/reviews?api_key=#{ENV['movie_api_key']}&language=en-US")
+  end
+
+  def movie_info(movie_id)
+    get_url("/movie/#{movie_id}?api_key=#{ENV['movie_api_key']}&language=en-US")
+  end
+
+  def top_rated
+    get_url("/movie/top_rated?api_key=#{ENV['movie_api_key']}&language=en-US")[:results]
+  end
+
+  def get_url(url)
+    response = Faraday.get("https://api.themoviedb.org/3#{url}")
```

Comment:
we coudl refactor to add the api key here. I don't think it matters what order the query parameters are in , so long as it's after the. '?'. We could add the api= param to the end of every url and remove that section from the other definitions.

Reply:
That's a good point! I hadn't thought of if the order matters or not.